module-4/backend/week_three.md

```diff
@@ -5,15 +5,42 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 Note: When you're done, submit a PR. 
 
 1. At a high level, what is Node?
+  - Node is an interpreter that is used to run Javascript code on a server.
 2. What is Express? What is Express similar to in the Ruby world?
+  - Express is middleware, and it is similar to rack in the Ruby world.
 3. How do we setup a route when creating an API with Node and Express? Please provide a code snippet.
+  ```javascript
+  let knex = require('./config');
+  let express = require('express')
+
+  app.get('/', function (req, res) {
+    database.raw('SELECT * FROM foods).then(
+      let someSqlQuery = data.rows
+    ).catch(
+      return "there was an error"
+     )
+    res.send(someSqlQuery)
+  })
+  ```
 4. What do we use Knex for?
+  Knex is used as an ORM to simplify SQL queries within an express application.  
 5. How **could** you organize your code to follow the MVC design pattern in your Quantified Self project?
+  You can create models and a controller.  The controller would handle the incoming web requests, while the models would handle operations such as quering a model's data tables.
 6. How do you execute raw SQL in node?
+  Knex supports raw sql queries with the .raw method that you can pass a raw SQL query to.
 7. What are some advantages to having your front end and back end code live in separate applications? What are some disadvantages?
+  An advantage to having them live in seperate applications is to have, at a high level, a clear seperation of responsibilities within your code.  It also allows work to be done concurrently by two or more developers since the two components are unrelated.  
+  A disadvantage is that there may be a disconnect between what the backened is doing and what the front end is doing and what decisions are made on each application.  A good example is to consider what type of data is returned from an API.  This data can be reformatted on either the front end side or the back end side, but if a developer is working on the front end, it may be tempting to change the front end behavior to reformat the data instead of change the back end behavior to serve the data in the correct format in the frist place.
 
 #### Review  
 
 8. Describe DNS.
+  DNS is used to resolve a host name to an IP address so that a client machine can send HTTP requests to a server.
```

Comment:
So that we can access something easily readible like "google.com" without needing to know its IP address